### PR TITLE
Add version and changelog to modules

### LINKED ISSUE #534: Improved module updates
~~safer updating without destroying students' work and/or~~ possibility to mark an update as "minor" so that people who have modified their copy of the module do not get an update notification

### PR DESCRIPTION
**Pull request by jaakkonarhi**
# Description of the PR

+ Replaces the id of modules with a version number major.minor
+ Version numbers are optional in the course config json and will default to 1.0
+ If the major number doesn't change between module versions, a notification and the update available text will only be shown if there are no local changes
+ Adds an optional changelog parameter to the modules which is shown in the tooltip if an update is available
+ Closes #534

# Testing
<table>
  <tr>
    <th>unit</th>
    <th>integration</th>
    <th>e2e</th>
    <th>manual</th>
  </tr>
  <tr>
    <td>
      <ul>
        <li>- [x] new <b>unit</b> tests created</li>
        <li>- [x] all <b>unit</b> tests pass</li>
      </ul>
    </td>
    <td>
      <ul>
        <li>- [ ] new <b>integration</b> tests created</li>
        <li>- [x] all <b>integration</b> tests pass</li>
      </ul>
    </td>
    <td>
      <ul>
        <li>- [ ] new <b>e2e</b> tests created</li>
        <li>- [x] all <b>e2e</b> tests pass</li>
      </ul>
    </td>
    <td>
      <ul>
        <li>- [x] <b>manual</b> testing went well</li>
      </ul>
    </td>
  </tr>
</table>  
  
# Have you updated the [TESTING.md](https://github.com/Aalto-LeTech/intellij-plugin/blob/master/TESTING.md) or other relevant documentation on _your_ branch?

- [ ] Yes.
- [ ] Not yet. I will do it next.
- [x] Not relevant.

### CODE REVIEW

File: src/main/java/fi/aalto/cs/apluscourses/presentation/module/ModuleListElementViewModel.java [hunk 1]
```diff
@@ -35,17 +34,21 @@ public String getUrl() {
...
-    return timestamp != null
-        ? PluginResourceBundle.getAndReplaceText(
-                "presentation.moduleTooltip.timestamp",
-                timestamp.format(DateTimeFormatter.ofLocalizedDateTime(FormatStyle.SHORT)))
-        : PluginResourceBundle.getAndReplaceText(
-                "presentation.moduleTooltip.moduleURL", getUrl());
+    var timestamp = getModel().getMetadata().getDownloadedAt();
+    var changelog = getModel().getChangelog();
+    if (isUpdateAvailable() && !changelog.equals("")) {
+      return "<html>What's new:<br>" + changelog + "</html>";
```
Comment (stellatsv) [inline src/main/java/fi/aalto/cs/apluscourses/presentation/module/ModuleListElementViewModel.java:46]: This String could be externalized.

File: src/main/java/fi/aalto/cs/apluscourses/presentation/module/ModuleListElementViewModel.java [hunk 2]
```diff
@@ -35,17 +34,21 @@ public String getUrl() {
...
+    } else if (timestamp != null) {
+      return PluginResourceBundle.getAndReplaceText("presentation.moduleTooltip.timestamp",
```
Comment (stellatsv) [inline src/main/java/fi/aalto/cs/apluscourses/presentation/module/ModuleListElementViewModel.java:48]: Nit: importing `PluginResourceBundle.getAndReplaceText` as static would make this a bit less verbose.

**Review by stellatsv — APPROVED** [review]
Good work!

**Comment by jaakkonarhi** [comment]
I think we should go with a single Version class, since it makes things simpler for now

**Pull request merged**